src/Compilers/CSharp/Portable/Symbols/Synthesized/Records/SynthesizedRecordCopyCtor.cs

```diff
@@ -58,19 +58,46 @@ internal override void GenerateMethodBodyStatements(SyntheticBoundNodeFactory F,
 
         internal static MethodSymbol? FindCopyConstructor(NamedTypeSymbol containingType, NamedTypeSymbol within, ref HashSet<DiagnosticInfo> useSiteDiagnostics)
         {
-            // We should handle ambiguities once we consider custom modifiers, as we do in overload resolution
-            // https://github.com/dotnet/roslyn/issues/45077
+            MethodSymbol? bestCandidate = null;
+            int bestModifierCountSoFar = -1; // stays as -1 unless we hit an ambiguity
             foreach (var member in containingType.InstanceConstructors)
             {
                 if (HasCopyConstructorSignature(member) &&
                     !member.HasUnsupportedMetadata &&
                     AccessCheck.IsSymbolAccessible(member, within, ref useSiteDiagnostics))
                 {
-                    return member;
+                    // If one has fewer custom modifiers, that is better
+                    // (see OverloadResolution.BetterFunctionMember)
+
+                    if (bestCandidate is null && bestModifierCountSoFar < 0)
+                    {
+                        bestCandidate = member;
+                        continue;
+                    }
+
+                    if (bestModifierCountSoFar < 0)
+                    {
+                        bestModifierCountSoFar = bestCandidate.CustomModifierCount();
+                    }
+
+                    var memberModCount = member.CustomModifierCount();
+                    if (bestModifierCountSoFar >= 0 && memberModCount > bestModifierCountSoFar)
+                    {
+                        continue;
+                    }
+
+                    if (memberModCount == bestModifierCountSoFar)
+                    {
+                        bestCandidate = null;
+                        continue;
+                    }
+
+                    bestCandidate = member;
+                    bestModifierCountSoFar = memberModCount;
```

Comment:
>bestModifierCountSoFar = memberModCount [](start = 20, length = 39)

`bestModiferCountSoFar` is only a short cut if there are 3 copy constructors which is pretty unlikely. It might be simpler to always call `CustomModiferCount()`.
```C#
if (bestCandidate.CustomModifierCount() > member.CustomModifierCount())
{
    bestCandidate = member;
}
```

Reply:
In a scenario with 3 overload, the first two having the same modifier count, we need to remember what that modifier count was when we consider the last overload. We need that state.